packages/core-p2p/src/peer-verifier.ts

```diff
@@ -0,0 +1,540 @@
+import { app } from "@arkecosystem/core-container";
+import { ConnectionInterface } from "@arkecosystem/core-database";
+import { PostgresConnection } from "@arkecosystem/core-database-postgres";
+import { Logger } from "@arkecosystem/core-interfaces";
+import { CappedSet, roundCalculator } from "@arkecosystem/core-utils";
+import { models } from "@arkecosystem/crypto";
+import assert from "assert";
+import { Peer } from './peer';
+
+/**
+ * A cache of verified blocks' ids. A block is verified if it is connected to a chain
+ * in which all blocks (including that one) are signed by the corresponding delegates.
+ */
+const verifiedBlocks = new CappedSet();
```

Comment:
Wondering why `verifiedBlocks` is here as a global variable and not as a property of the class ?

Reply:
My intention is to have it shared between all instances of the class. For example - if we verify blockX and blockY during verifying peer1, and after this we start verifying peer2 and he says he has blockX, then we would not need to download blockX from peer2 and run it through verification.

I wonder if there is a better way to have the `verifiedBlocks` shared between the class instances? If not maybe at least I should put a comment explaining the intention.

Reply:
You can make it private static field inside the class e.g 
```typescript
private static readonly verifiedBlocks = new CappedSet();
```

and then access it from within an instance via `PeerVerifier.verifiedBlocks`.

Reply:
Ok thanks resolving conversation, will run it through my tests now.